test/lib/skunk/application_test.rb

```diff
@@ -27,5 +28,26 @@
         _(result).must_equal success_code
       end
     end
+
+    context "when passing --out option with a file" do
+      require "fileutils"
+
+      let(:argv) { ["--out=tmp/generated_report.txt", "samples/rubycritic"] }
+      let(:success_code) { 0 }
+
+      it "writes output to the file" do
+        FileUtils.rm("tmp/generated_report.txt", force: true)
+        FileUtils.mkdir_p("tmp")
+
+        RubyCritic::AnalysedModule.stub_any_instance(:churn, 1) do
+          RubyCritic::AnalysedModule.stub_any_instance(:coverage, 100) do
+            result = application.execute
+            _(result).must_equal success_code
+          end
+        end
+
+        _(File.read("tmp/generated_report.txt")).must_equal File.read("test/samples/console_output.txt")
```

Comment:
@etagwerker @bronzdoc I've been banging my head with this assertion, I can't make pass. 

I'm not being able to read any contents from `tmp/generated_report.txt`, they are consistently empty (`""`). 

After the tests finish I can confirm the contents are flushed to the file, but not within the test. This is weird. I tried doing `#flush` on the output stream without success. I also tried using the `fsync` method of `IO`, but nothing.

Do you have any ideas? 🦆

Reply:
looking into it

Reply:
@bronzdoc what happened with those commits, I noticed the history got rewritten on this branch. Did you do a rebase against master?  Next time please let's get on the same page before doing commits that break history on branches we might be sharing to avoid conflicts.

Regarding the assertion that wasn't passing on my test, did you figure an explanation for why it wasn't working? I noticed test are green now.

Reply:
@manuca what commits? the commits are there. I just removed one commit that I made that was not needed. Beside that I fixed the merge conflicts and a force push had to be made.

I think the issue was that we were returning a stream/file in the method but we never closed it. So i guess while is opened we cannot read it.

```
def output_stream
  output_filename = @argv_options.output_filename
  output_filename.nil? ? $stdout : File.open(output_filename, "w")
end
```
everytime we used `output_stream` it will return a new file object so even if we do something like this:
```ruby
output_stream.puts "hello world"
output_stream.close
```
that will not close the file created the first time when we do `output_stream.puts "hello world"`

Reply:
I tried closing the output stream explicitly when we were handling output to a file as well as flushing the stream but couldn't make it work either way so I didn't think it had to do with that, very strange issue.